tests/storage/scsidisk.go

```diff
@@ -0,0 +1,184 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// The tests using the function CreateErrorDisk need to be run serially as it relies on the kernel scsi_debug module
+func CreateErrorDisk(nodeName string) (address string, device string) {
+	By("Creating error disk")
+	return CreateSCSIDisk(nodeName, []string{"opts=2", "every_nth=4", "dev_size_mb=8"})
+}
+
+// CreateSCSIDisk creates a SCSI disk using the scsi_debug module. This function should be used only to check SCSI disk functionalities and not for creating a filesystem or any data. The disk is stored in ram and it isn't suitable for storing large amount of data.
+// If a test uses this function, it needs to be run serially. The device is created directly on the node and the addition and removal of the scsi_debug kernel module could create flakiness
+func CreateSCSIDisk(nodeName string, opts []string) (address string, device string) {
+	args := []string{"/usr/sbin/modprobe", "scsi_debug"}
+	args = append(args, opts...)
+	_, err := virtChrootExecuteCommandInVirtHandlerPod(nodeName, args)
+	ExpectWithOffset(1, err).NotTo(HaveOccurred(), "Failed to create faulty disk")
+
+	EventuallyWithOffset(1, func() error {
+		args = []string{"/bin/sh", "-c", "/bin/grep -l scsi_debug /sys/bus/scsi/devices/*/model"}
+		stdout, err := virtChrootExecuteCommandInVirtHandlerPod(nodeName, args)
+		if err != nil {
+			return err
+		}
+
+		// Example output
+		// /sys/bus/scsi/devices/0:0:0:0/model
+		if !filepath.IsAbs(stdout) {
+			return fmt.Errorf("Device path extracted from sysfs is not populated: %s", stdout)
+		}
+
+		pathname := strings.Split(stdout, "/")
+		address = pathname[5]
+
+		args = []string{"/bin/ls", "/sys/bus/scsi/devices/" + address + "/block"}
+		stdout, err = virtChrootExecuteCommandInVirtHandlerPod(nodeName, args)
+		if err != nil {
+			return err
+		}
+		device = "/dev/" + strings.TrimSpace(stdout)
+
+		return nil
+	}, 20*time.Second, 5*time.Second).ShouldNot(HaveOccurred())
+
+	return address, device
+}
+
+func RemoveSCSIDisk(nodeName, address string) {
```

Comment:
I'd also suggest to unxeport these functions if they're only used in the storage package, but I won't require it for the scope of this PR.

Reply:
Thanks, @acardace !
I hope @0xFelix can express that this is good enough for him, too.

Reply:
Yes, that can be done but really this PR is too good to keep holding for this.
I think it makes sense to follow up later